modules/core/src/adapter/device.ts

```diff
@@ -267,7 +267,7 @@ export abstract class Device {
     id: null!,
     powerPreference: 'high-performance',
     failIfMajorPerformanceCaveat: false,
-    canvasContext: undefined!,
+    createCanvasContext: undefined!,
```

Comment:
Does WebGL require a canvas context? If so, should createCanvasContext default to true? Or is there some protection in place for WebGL to always create a canvas context?

Reply:
I think that's reasonable